code/src/main/java/com/onlinecontacttracing/servlets/get-data/CalendarDataForPositiveUser.java

```diff
@@ -59,18 +85,40 @@ public void run() {
       
       // Iterate through events to extract contacts and places
       for (Event event : events.getItems()) {
-        List<EventAttendee> attendees = Optional.ofNullable(event.getAttendees()).orElse(Collections.emptyList());
-
-        for (EventAttendee attendee : attendees) {
-          contacts.add(new PotentialContact(attendee.getDisplayName(), attendee.getEmail()));
-        }
+        getContactsFromEvent(event);
+        getPlacesFromEvent(event, context);
       }
 
-      // TODO: add positiveUserLocations
+      // Store data or replace old data with newer data.
+      ofy.save().entity(positiveUserPlaces).now();
       
     } catch (Exception e) {
       e.printStackTrace();
       log.warning("An exception occurred: " + e.toString());
     }
   }
+
+  private void getContactsFromEvent(Event event) {
+    List<EventAttendee> attendees = Optional.ofNullable(event.getAttendees()).orElse(Collections.emptyList());
+
+    for (EventAttendee attendee : attendees) {
```

Comment:
consider using a stream here instead, using map(...)

Reply:
for sure